### PR TITLE
Add generative model metrics notebook

### LINKED ISSUE #1770: realism_diversity_metrics

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
Fixes #1770  .

### Description
A few sentences describing the changes proposed in this pull request.

### Checks

- [ ] Avoid including large-size files in the PR.
- [ ] Clean up long text outputs from code cells in the notebook.
- [ ] For security purposes, please check the contents and remove any sensitive info such as user names and private key.
- [ ] Ensure (1) hyperlinks and markdown anchors are working (2) use relative paths for tutorial repo files (3) put figure and graphs in the `./figure` folder
- [ ] Notebook runs automatically `./runner.sh -t <path to .ipynb file>`

### CODE REVIEW

**Comment by yiheng-wang-nv** [comment]
Hi @ericspod @KumoLiu 
the model weights used in this tutorial is removed, and the url mentioned in the original repo does not exist:
https://github.com/marksgraham/pretrained_generative_models/blob/main/models/ddpm_2d.py#L34

Do you have any suggestions? Can we use other weights to replace it?
The cell shows:

```
if use_pre_trained:
    unet = torch.hub.load("marksgraham/pretrained_generative_models:v0.2", model="ddpm_2d", verbose=True)
    unet = unet.to(device)
else:
    model_path = Path.cwd() / Path("tutorials/generative/2d_ldm/best_aeutoencoderkl.pth")
    autoencoderkl.load_state_dict(torch.load(str(model_path)))
    model_path = Path.cwd() / Path("tutorials/generative/2d_ldm/best_unet.pth")
    unet.load_state_dict(torch.load(str(model_path)))
```
it seems the 2d_ldm weights can be used, can we prepare a copy for users to download?

**Comment by ericspod** [comment]
@marksgraham would you have the file that was stored in your google drive? We can rehost it somewhere else perhaps and get this added.

**Comment by marksgraham** [comment]
Sorry I deleted them in a bulk drive clearout, without realising they were still being used here.

It is easy to regenerate the checkpoint by running the tutorial with `use_pretrained=False`. Then it can be stored somewhere more permanent..

**Comment by yiheng-wang-nv** [comment]
> Sorry I deleted them in a bulk drive clearout, without realising they were still being used here.
> 
> It is easy to regenerate the checkpoint by running the tutorial with `use_pretrained=False`. Then it can be stored somewhere more permanent..

Hi @marksgraham , thanks for the info. Can I confirm that we at least need to use trained `2d_ldm` weights?

**Comment by marksgraham** [comment]
> > Sorry I deleted them in a bulk drive clearout, without realising they were still being used here.
> > It is easy to regenerate the checkpoint by running the tutorial with `use_pretrained=False`. Then it can be stored somewhere more permanent..
> 
> Hi @marksgraham , thanks for the info. Can I confirm that we at least need to use trained `2d_ldm` weights?

Hi @yiheng-wang-nv yes they will need to be trained too. You should be able to get trained versions for both of the sets of weights using [this tutorial](https://github.com/Project-MONAI/tutorials/blob/1cd41fc23ec4f01826a86f9973bda51646ac6418/generation/2d_ldm/2d_ldm_tutorial.ipynb)

**Comment by yiheng-wang-nv** [comment]
Thanks @marksgraham , I will run 2d_ldm tutorial, and save weights into https://github.com/Project-MONAI/MONAI-extra-test-data, then we can use that links in the metric tutorial. cc @ericspod @KumoLiu

File: generation/realism_diversity_metrics/realism_diversity_metrics.ipynb [hunk 1]
```diff
@@ -0,0 +1,811 @@
+{
```
Comment (KumoLiu) [inline generation/realism_diversity_metrics/realism_diversity_metrics.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    unet_url = "https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/tutorial_generation_unet.pth"
</pre><p>To avoid confusion with maisi later, please help modify the weight path to include "2d_ldm".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/1771/discussion/#comment-1718243923'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

**Comment by yiheng-wang-nv** [comment]
Hi @ericspod , do you have any experiences on this error: https://github.com/Project-MONAI/tutorials/actions/runs/10400120252/job/28800115713?pr=1771#step:5:710
Run the notebook locally does not bring me same error.

**Comment by yiheng-wang-nv** [comment]
> Hi @ericspod , do you have any experiences on this error: https://github.com/Project-MONAI/tutorials/actions/runs/10400120252/job/28800115713?pr=1771#step:5:710 Run the notebook locally does not bring me same error.

I guess the CI machine may have some issues when load radimagenet weights, I changed to use torchvision's resnet50 as the feature extractor, and now tests pass.

**Comment by KumoLiu** [comment]
> I guess the CI machine may have some issues when load radimagenet weights, I changed to use torchvision's resnet50 as the feature extractor, and now tests pass.

I think the update here should be fine since the notebook is mainly talk about the metric used in the generative model. What do you think? @ericspod

**Comment by ericspod** [comment]
> Hi @ericspod , do you have any experiences on this error: https://github.com/Project-MONAI/tutorials/actions/runs/10400120252/job/28800115713?pr=1771#step:5:710
> Run the notebook locally does not bring me same error.

If you can get it to work with another network let's go with that. For the future this is related to numerical error with `scipy.linalg.sqrtm` in the FID calculation that's somewhat documented in other implementations, such as [here](https://stackoverflow.com/questions/77427606/fid-error-imaginary-component-in-google-colab-pytorch).  It appears to be sensitive to the version of scipy being used or other libraries it relies on.

File: generation/realism_diversity_metrics/realism_diversity_metrics.ipynb [hunk 2]
```diff
@@ -0,0 +1,841 @@
...
```
Comment (KumoLiu) [inline generation/realism_diversity_metrics/realism_diversity_metrics.ipynb:1]: <p>Please help update the description about "RadImageNet" here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/1771/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

**Review by KumoLiu — APPROVED** [review]

File: generation/realism_diversity_metrics/realism_diversity_metrics.ipynb (hunk 2 above)
Comment (yiheng-wang-nv) [reply generation/realism_diversity_metrics/realism_diversity_metrics.ipynb:1]: thanks, I removed the `notes` but keep the reference (since we still mentions this network)
